dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/DataAccessManager.kt

```diff
@@ -0,0 +1,179 @@
+            )?.any { it.accessStatus == AccessStatus.Granted } ?: false
+
+        if (hasAccess) {
+            accessRequestLogger.logMessageForCheckingIfUserHasAccessToDataset(
+                companyId,
+                dataType,
+                reportingPeriod,
+            )
+        }
+        return hasAccess
+    }
+
+    /**
+     * This method checks if the requesting user has granted access to the information for a specific company,
+     * reporting period and data type.
+     * If there is no access or no data set an exception is thrown.
+     * @param companyId the companyId for which the access status should be checked
+     * @param reportingPeriod the reportingPeriod for which the access status should be checked
+     * @param dataType the framework dataType for which the access status should be checked
+     * @param userId the userId for which the access status should be checked
+     */
+    @Transactional
+    fun headAccessToDataset(
+        companyId: String,
+        reportingPeriod: String,
+        dataType: String,
+        userId: String,
+    ) {
+        val dataTypeEnum = DataTypeEnum.decode(dataType)
+            ?: throw InvalidInputApiException(
+                "The provided input did not match expected values.",
+                "The $dataType was not recognized by the system. Please check your input",
+            )
+
+        if (dataTypeEnum != DataTypeEnum.vsme) {
+            return
+        }
+
+        val hasAccess =
+            hasAccessToPrivateDataset(companyId, reportingPeriod, dataTypeEnum, userId)
+
+        if (!hasAccess) {
+            throw ResourceNotFoundApiException(
+                "The user has no access to the dataset or the dataset does not exists.",
+                "The user $userId cannot access the dataset for the company $companyId, for the data type " +
+                    "$dataType and the reporting period $reportingPeriod. The dataset may not exists.",
+            )
+        }
+    }
+
+    /**
+     * This method is used to request access to a private dataset
+     * @param userId the userId of the user requesting access to the dataset
+     * @param companyId the companyId of the company to which the dataset belongs
+     * @param dataType the datatype of the dataset to which access was requested
+     * @param reportingPeriod the reportingPeriod of the dataset to which access was requested
+     */
+    fun createAccessRequestToPrivateDataset(
+        userId: String,
+        companyId: String,
+        dataType: DataTypeEnum,
+        reportingPeriod: String,
+        contacts: Set<String>?,
+        message: String?,
+    ) {
+        val existingRequestsOfUser = dataRequestRepository
+            .findByUserIdAndDatalandCompanyIdAndDataTypeAndReportingPeriod(
+                userId, companyId, dataType.name, reportingPeriod,
+            )
+        if (!existingRequestsOfUser.isNullOrEmpty()) {
+            val dataRequestEntity = existingRequestsOfUser[0]
```

Comment:
existingRequestsOfUser.first() instead of direct index accessing may be preferrable kotlin code